### PR TITLE
Workspace minimap

### PR DESCRIPTION
**Pull request by yoching**
Hello, I've made a minimap in Workspace!

I think I've made the most of the functionalities, however, there are a few remaining things:

- Toggle the visibility of the map?
  Currently, some underlying elements are hidden and the close button ("x") in WorkspaceItem is not clickable when it's under the minimap. Do you have any direction on how to solve this issue?
- Smaller screens (e.g.mobile)
   Is it better to hide the minimap on smaller screens?
- I hide the keyboard shortcuts view for now, because it seems too much for one task. 

As my CSS experience is a bit limited, any suggestions about styling are welcome.

Thank you in advance!

Screenshot 2023-05-23 at 16 44 31

### CODE REVIEW

**Comment by yoching** [comment]
Related issue: https://github.com/unisonweb/ui-core/issues/7#issuecomment-1518656428

**Review by hojberg — COMMENTED** [review]
Really awesome!

I added a few notes, mostly stylistic stuff that we could probably add to a contributions file or something because there's really no way you would have known.

Thank you for all this effort! Its really cool.

For mobile and smaller viewports I think we want to do something similar to what we do with the sidebar:

Screenshot 2023-05-24 at 1 01 45 PM

That Icon is `Icon.unfoldedMap`.

This is also in a darker color scheme, which I recently landed on, but we don't have to change this PR for either of those I think.

File: src/Code/Workspace.elm [hunk 1]
```diff
@@ -270,6 +272,33 @@ update config viewMode msg ({ workspaceItems } as model) =
...
+            let
+                nextWorkspaceItems =
+                    case mMsg of
+                        WorkspaceMinimap.SelectItem item ->
+                            item
+                                |> WorkspaceItem.reference
+                                |> WorkspaceItems.focusOn model.workspaceItems
+
+                        WorkspaceMinimap.CloseAll ->
+                            WorkspaceItems.empty
```
[the remark this answers is not in the document]
Comment (yoching) [reply src/Code/Workspace.elm:285]: Sounds interesting. I'll try this approach

File: src/css/code/workspace-minimap.css [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    padding-left: 0.3rem;
+    padding-right: 0.3rem;
+}
+
+.workspace-minimap-entry .workspace-minimap-entry-content .category-icon {
+    display: inline-flex;
+    align-items: center;
+    justify-content: center;
+    flex-direction: row;
+    width: 1.5rem;
```
[the remark this answers is not in the document]
Comment (yoching) [reply src/css/code/workspace-minimap.css:51]: This was a copy from workspace-item.css. Is it better to use font-size here?
https://github.com/unisonweb/ui-core/blob/main/src/css/code/workspace-item.css#L136-L141

File: src/Code/Workspace.elm (hunk 1 above)
[the remark this answers is not in the document]
Comment (yoching) [reply src/Code/Workspace.elm]: I updated with this approach.

File: src/Code/Workspace.elm [hunk 3]
```diff
@@ -520,7 +549,14 @@ view viewMode model =
             case viewMode of
                 ViewMode.Regular ->
                     article [ id "workspace", class (ViewMode.toCssClass viewMode) ]
-                        [ section
+                        [ aside
+                            [ id "workspace-minimap" ]
```
[the remark this answers is not in the document]
Comment (yoching) [reply src/Code/Workspace.elm]: Updated🙂

File: src/Code/Workspace/WorkspaceMinimap.elm [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+                    [ div
+                        [ class "name" ]
+                        [ text "Loading" ]
+                    ]
+
+                Failure _ _ ->
+                    [ div
+                        [ class "name" ]
+                        [ text "Failure" ]
+                    ]
```
[the remark this answers is not in the document]
Comment (yoching) [reply src/Code/Workspace/WorkspaceMinimap.elm]: I added a placeholder and status banner. Do they seem good?

File: src/css/code/workspace-minimap.css [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+.workspace-minimap-entry.focused {
+    background-color: var(--color-blue-5);
+}
+
+.workspace-minimap-entry .workspace-minimap-entry-content {
+    display: flex;
+    align-items: center;
+    padding-left: 0.3rem;
+    padding-right: 0.3rem;
```
[the remark this answers is not in the document]
Comment (yoching) [reply src/css/code/workspace-minimap.css]: Updated

File: src/css/code/workspace-minimap.css [hunk 6]
```diff
@@ -0,0 +1,65 @@
+.workspace-minimap-content {
+    min-width: 10rem;
+    background-color: white;
+    padding: 1rem;
+    border-width: 1px;
+    border-color: var(--color-gray-lighten-50);
+    border-style: solid;
+    border-radius: 5px;
```
[the remark this answers is not in the document]
Comment (yoching) [reply src/css/code/workspace-minimap.css]: Updated

File: src/css/code/workspace-minimap.css [hunk 7]
```diff
@@ -0,0 +1,65 @@
+.workspace-minimap-content {
+    min-width: 10rem;
+    background-color: white;
```
[the remark this answers is not in the document]
Comment (yoching) [reply src/css/code/workspace-minimap.css]: used `--u-color_container`

File: src/css/code/workspace-minimap.css [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+    align-items: center;
+    justify-content: center;
+    flex-direction: row;
+    width: 1.5rem;
+    line-height: 1;
+}
+
+.workspace-minimap-entry .keyboard-shortcut {
+    padding-right: 0.5rem;
+}
```
[the remark this answers is not in the document]
Comment (yoching) [reply src/css/code/workspace-minimap.css:57]: Changed the indentation!

**Comment by yoching** [comment]
Added minimap toggle functionality

**Comment by yoching** [comment]
Tweaked CSS a bit.

Screenshot 2023-06-09 at 12 35 07

Screenshot 2023-06-09 at 12 35 11

Screenshot 2023-06-09 at 12 35 23

**Comment by yoching** [comment]
Updated some reviews and styles, fixed conflicts✨

**Comment by yoching** [comment]
Yes, please try on them if possible!

**Pull request merged**